If there is a parsing error, it is due to the failure to determine whether BGPMessage is null, leading to an invalid memory address or nil pointer dereference

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1a pc=0xa76016]
goroutine 11071628 [running]:
github.com/osrg/gobgp/v3/pkg/server.(*fsmHandler).handlingError(0xc024d8cee8, 0xc0006de1c0, {0xda4ca0, 0xc00b6d5bc0}, 0x58)
/builds/gobgp/pkg/server/fsm.go:886 +0x36
github.com/osrg/gobgp/v3/pkg/server.(*fsmHandler).recvMessageWithError(0xc007ca6540)
/builds/gobgp/pkg/server/fsm.go:990 +0x5b3
github.com/osrg/gobgp/v3/pkg/server.(*fsmHandler).recvMessage(0xc007ca6540, {0xdb7a58, 0xc01c6ed900}, 0xc0083f0340)
/builds/gobgp/pkg/server/fsm.go:1147 +0x6d
created by github.com/osrg/gobgp/v3/pkg/server.(*fsmHandler).opensent
/builds/gobgp/pkg/server/fsm.go:1231 +0x2c7

```

https://github.com/osrg/gobgp/blob/c3eb0db6df91379e15ce1a8709cb1b64f482248e/pkg/server/fsm.go#L1311C16-L1311C28

```

func ParseBGPBody(h *BGPHeader, data []byte, options ...*MarshallingOption) (*BGPMessage, error) {
	return parseBody(h, data, options...)
}

func parseBody(h *BGPHeader, data []byte, options ...*MarshallingOption) (*BGPMessage, error) {
	if len(data) < int(h.Len)-BGP_HEADER_LENGTH {
		return nil, NewMessageError(BGP_ERROR_MESSAGE_HEADER_ERROR, BGP_ERROR_SUB_BAD_MESSAGE_LENGTH, nil, "Not all BGP message bytes available")
	}
...
}


m, err := bgp.ParseBGPBody(hd, bodyBuf, &bgp.MarshallingOption{
		AddPath:    h.fsm.familyMap.Load().(map[bgp.Family]bgp.BGPAddPathMode),
		Use2ByteAS: h.fsm.twoByteAsTrans, // true if peer does NOT support 4-byte AS
	})
if err != nil {
	handling = h.handlingError(m, err, useRevisedError)
}


func (h *fsmHandler) handlingError(m *bgp.BGPMessage, e error, useRevisedError bool) bgp.ErrorHandling {
	
// *** panic caused by m nil***
	if m.Header.Type == bgp.BGP_MSG_UPDATE && useRevisedError {
}
}
```


***If there is a parsing error, it is due to the failure to determine whether BGPMessage is null, leading to an invalid memory address or nil pointer dereference***

--- a/pkg/server/fsm.go
+++ b/pkg/server/fsm.go
@@ -1313,7 +1313,11 @@ func (h *fsmHandler) recvMessageWithError(conn net.Conn, stateReasonCh chan<- fs
 		Use2ByteAS: h.fsm.twoByteAsTrans, // true if peer does NOT support 4-byte AS
 	})
 	if err != nil {
-		handling = h.handlingError(m, err, useRevisedError)
+		if m == nil {
+			handling = bgp.ERROR_HANDLING_SESSION_RESET
+		} else {
+			handling = h.handlingError(m, err, useRevisedError)
+		}
 		h.fsm.bgpMessageStateUpdate(0, true)
 	} else {
 		h.fsm.bgpMessageStateUpdate(m.Header.Type, true)

--- a/pkg/server/fsm_test.go
+++ b/pkg/server/fsm_test.go
@@ -1319,6 +1319,39 @@ func TestRecvMessageWithError_MalformedNextHop(t *testing.T) {
 	})
 }
 
+// TestRecvMessageWithError_UnknownMessageType verifies that recvMessageWithError
+// handles an unknown BGP message type without panicking on nil dereference.
+// This is a regression test for https://github.com/osrg/gobgp/issues/3325.
+func TestRecvMessageWithError_UnknownMessageType(t *testing.T) {
+	assert := assert.New(t)
+
+	m := NewMockConnection()
+	_, h := makePeerAndHandler(m)
+	t.Cleanup(func() {
+		h.outgoing.Close()
+		h.fsm.outgoingCh.Close()
+		h.fsm.conn.Close()
+	})
+
+	// BGP message with unknown type 0xFF.
+	// ParseBGPBody returns (nil, error) for unknown types.
+	raw := []byte{
+		0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+		0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, // marker
+		0x00, 0x13, // length: 19 (header only, no body)
+		0xff, // type: unknown
+	}
+
+	go m.remote.Write(raw)
+
+	stateReasonCh := make(chan fsmStateReason, 2)
+	fmsg, err := h.recvMessageWithError(m.Conn, stateReasonCh)
+
+	assert.Error(err)
+	assert.NotNil(fmsg)
+	assert.Equal(bgp.ERROR_HANDLING_SESSION_RESET, fmsg.handling)
+}
+
 // TestBMPStatsUpdate verifies that BMP stats are correctly updated via
 // atomic operations and exposed via toConfig.
 func TestBMPStatsUpdate(t *testing.T) {

